src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1526,7 +1530,13 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
         }
 
       if (deploy_has_bool_option (self, "reboot"))
-        rpmostreed_daemon_reboot (rpmostreed_daemon_get ());
+        {
+          GDBusConnection *connection =
+            rpmostreed_daemon_connection ();
+          if (!check_sd_inhibitor_locks (connection, cancellable, error))
```

Comment:
Seems simpler to me to have this be `if (!check_sd_inhibitor_locks(error)) return FALSE;` and do the `rpmostreed_daemon_connection()` inside that function.